README.md

```diff
@@ -124,29 +140,30 @@ org.freedesktop.tuhi1.Device
       Otherwise, the device is paired and this function returns success (0)
       or a negative errno on failure.
 
-  Method: Listen() -> ()
+  Method: StartListening() -> ()
       Listen for data from this device. This method starts listening for
-      events on the device for an unspecified timeout. When the timeout
-      expires, a ListenComplete signal is sent indicating success or error.
+      events on the device. Clients need to call StopListening() to
+      stop the events processing. If the client disconnects, StopListening()
+      is matched by the daemon.
 
-      This function requires the device to be connected and may require some
-      interactivity (e.g. the user may need to press the sync button).
+      This method effectively sets the property Listening to True.
 
-      When the device connects, the daemon downloads all drawings from the
-      device and disconnects from the device. If successfull, the drawings
-      are deleted from the device. The data is held by the daemon in
-      non-persistent storage until the daemon is stopped or we run out of
-      memory, whichever happens earlier.  Use GetJSONData() to retrieve the
-      data from the daemon.
+      A client can not call twice StartListening() on the same device without
+      a matching StopListening() in the middle. This will result in a failure.
```

Comment:
Is there a reason why we can't just let the client call `StartListening()` and ignore it once we're in progress? Same with `StopListening()` can just be a noop if we're not doing so already.

If we do want this to be a single call, we should return an error here instead of `()`


fwiw, I have a local FIXME commit here that fixes some of the grammar, will get to that tomorrow

Reply:
> Is there a reason why we can't just let the client call `StartListening()` and ignore it once we're in progress? Same with `StopListening()` can just be a noop if we're not doing so already.

I mostly copied/pasted the behaviour of the bluez stack regarding `Start/StopDiscovery`.

The question is actually more whther or not we want to enforce the clients to behave sanely (matching start and stop) or we will just be happy having clients repeatively call `StartListening` because they didn't read the spec.

> If we do want this to be a single call, we should return an error here instead of ()

Ack to that

> fwiw, I have a local FIXME commit here that fixes some of the grammar, will get to that tomorrow

cool :)